File: Algorithms/src/1769. Minimum Number of Operations to Move All Balls to Each Box/Code/solution.go
func minOperations(boxes string) []int {
    n := len(boxes)
    answer := make([]int, n)

    // Left-to-right pass
    balls, operations := 0, 0
    for i := 0; i < n; i++ {
        answer[i] += operations
        if boxes[i] == '1' {
            balls++ // Count balls
        }
        operations += balls // Add the current number of balls to operations
    }

    // Right-to-left pass
    balls, operations = 0, 0
    for i := n - 1; i >= 0; i-- {
        answer[i] += operations
        if boxes[i] == '1' {
            balls++ // Count balls
        }
        operations += balls // Add the current number of balls to operations
    }

    return answer
}

File: Algorithms/src/1769. Minimum Number of Operations to Move All Balls to Each Box/Code/solution.js
/**
 * @param {string} boxes
 * @return {number[]}
 */
var minOperations = function (boxes) {
  const n = boxes.length;
  const answer = new Array(n).fill(0);

  // Left-to-right pass
  let balls = 0,
    operations = 0;
  for (let i = 0; i < n; i++) {
    answer[i] += operations;
    balls += boxes[i] === "1" ? 1 : 0; // Count balls
    operations += balls; // Add the current number of balls to operations
  }

  // Right-to-left pass
  balls = 0;
  operations = 0;
  for (let i = n - 1; i >= 0; i--) {
    answer[i] += operations;
    balls += boxes[i] === "1" ? 1 : 0; // Count balls
    operations += balls; // Add the current number of balls to operations
  }

  return answer;
};
